sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -91,11 +91,7 @@ case class Exchange(
       shuffleManager.isInstanceOf[UnsafeShuffleManager]
     val bypassMergeThreshold = conf.getInt("spark.shuffle.sort.bypassMergeThreshold", 200)
     val serializeMapOutputs = conf.getBoolean("spark.shuffle.sort.serializeMapOutputs", true)
-    if (newOrdering.nonEmpty) {
-      // If a new ordering is required, then records will be sorted with Spark's `ExternalSorter`,
-      // which requires a defensive copy.
-      true
-    } else if (sortBasedShuffleOn) {
```

Comment:
Just try to understand the context. Is this `if (newOrdering.nonEmpty)` part not necessary because of our recent change?